go/runtime/bundle/manager.go

```diff
@@ -236,20 +247,44 @@ func (m *Manager) Queue(runtimeID common.Namespace, manifestHashes []hash.Hash)
 	}
 	m.downloadQueue[runtimeID] = hashes
 
-	// Trigger immediate download of new bundles.
+	// Trigger immediate download and clean-up of stale bundles.
 	select {
-	case m.downloadCh <- struct{}{}:
+	case m.triggerCh <- struct{}{}:
 	default:
 	}
 }
 
-// Download tries to download bundles in the queue.
-func (m *Manager) Download() {
+// Cleanup initiates the clean-up of regular bundles for the specified runtime,
+// removing versions lower than the specified one.
+func (m *Manager) Cleanup(runtimeID common.Namespace, version version.Version) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	m.cleanupQueue[runtimeID] = version
+
+	// Trigger immediate download and clean-up of stale bundles.
+	select {
+	case m.triggerCh <- struct{}{}:
+	default:
+	}
+}
+
+// download tries to download bundles in the queue.
+func (m *Manager) download() {
 	for runtimeID := range m.runtimeIDs {
 		m.downloadBundles(runtimeID)
 	}
 }
 
+// clean removes outdated manifest and deletes corresponding
```

Comment:
Comments for `Clean` and `clean` are quite different, which makes me wonder what is going on with manifests and when are they outdated, Maybe we just remove comments for `download` and `clean` .